src/caller.jl

```diff
@@ -0,0 +1,189 @@
+
+function recursive_translate(x, R)
+    if length(x) > 0 && x[1] isa Bool
+        return convert_return_list(x, R)
+    else
+        return [ recursive_translate(i, R) for i in x]
+    end
+end
+
+# function LIST_CMD_CASTER(x)
+#     x_new = LIST_CMD_TRAVERSAL(x)
+# end
+
+casting_functions_pre = Dict(:NUMBER_CMD     => (libSingular.NUMBER_CMD_CASTER,     true, ()),
+    :RING_CMD       => (libSingular.RING_CMD_CASTER,       false, ()),
+    :POLY_CMD       => (libSingular.POLY_CMD_CASTER,       true, ()),
+    :IDEAL_CMD      => (libSingular.IDEAL_CMD_CASTER,      true, ()),
+    :MODUL_CMD      => (libSingular.IDEAL_CMD_CASTER,      true, (:module,)),
+    :VECTOR_CMD     => (libSingular.POLY_CMD_CASTER,       true, (:vector,)),
+    :INT_CMD        => (libSingular.INT_CMD_CASTER,        false, ()),
+    :STRING_CMD     => (libSingular.STRING_CMD_CASTER,     false, ()),
+    :LIST_CMD       => (libSingular.LIST_CMD_TRAVERSAL,    false, ()),
+    :INTVEC_CMD     => (libSingular.INTVEC_CMD_CASTER,     false, ()),
+    :INTMAT_CMD     => (libSingular.INTMAT_CMD_CASTER,     false, ()),
+    :BIGINT_CMD     => (libSingular.BIGINT_CMD_CASTER,     false, ()),
+    :BIGINTMAT_CMD  => (libSingular.BIGINTMAT_CMD_CASTER,  false, ()),
+    :MAP_CMD        => (libSingular.MAP_CMD_CASTER,        false, ()),
+    :RESOLUTION_CMD => (libSingular.RESOLUTION_CMD_CASTER, true, (:resolution,)))
+
+casting_functions = nothing
+
+function create_casting_functions()
+    pair_array = Any[]
+    for (sym, func) in casting_functions_pre
+        push!(pair_array, mapping_types_reversed[sym] => func)
+    end
+    return Dict(pair_array...)
+end
+
+function convert_ring_content(value_list, rng)
+    return_dict = Dict{Symbol,Any}()
+    for i in value_list
+        return_dict[i[2]] = convert_return_value([false,i[3],i[1]], rng)
+    end
+    return return_dict
+end
+
+function convert_return_value(single_value, rng = nothing)
```

Comment:
Some comments here and elsewhere are badly missing. For starters, what is `single_value` expected to contain?